indico/modules/events/registration/fields/simple.py

```diff
@@ -311,22 +310,26 @@ def process_formdata(self, valuelist):
 
 class FileField(RegistrationFormFieldBase):
     name = 'file'
-    wtf_field_class = _DeletableFileField
+    # Toggle this to make the angular form work
+    # wtf_field_class = _DeletableFileField
+    wtf_field_class = wtforms.StringField
 
     def process_form_data(self, registration, value, old_data=None, billable_items_locked=False):
         data = {'field_data': self.form_item.current_data}
         if not value:
+            data['file'] = None
             return data
-        file_ = value['uploaded_file']
-        if file_:
+
+        if file := File.query.filter(File.uuid == value, ~File.claimed).first():
             # we have a file -> always save it
             data['file'] = {
-                'data': file_.stream,
-                'name': secure_client_filename(file_.filename),
-                'content_type': mimetypes.guess_type(file_.filename)[0] or file_.mimetype or 'application/octet-stream'
+                'data': file.open(),
```

Comment:
You could replace this whole block with `data['file'] = file` and then handle this in `_set_file` in `RegistrationData`.

That way you can properly do `with file.open() as f: ...` instead of opening it here and never closing it.